crates/wash-cli/src/up/mod.rs

```diff
@@ -494,17 +501,65 @@ pub async fn handle_up(cmd: UpCommand, output_kind: OutputKind) -> Result<Comman
 
     spinner.finish_and_clear();
 
+    // Start building the CommandOutput providing some useful information like pids, ports, and logfiles
+    let mut out_json = HashMap::new();
+    let mut out_text = String::from("");
+    out_json.insert("success".to_string(), json!(true));
+    out_text.push_str("🛁 wash up completed successfully");
+
+    // If a WADM manifest was provided, run it
+    if let Some(ref manifest_path) = cmd.wadm_opts.wadm_application {
+        out_text.push_str("⏲  Waiting for host start to deploy WADM application...");
+
+        wait_for_host(cmd.detached, &wasmcloud_log_path, &mut wasmcloud_child)
+            .await
+            .context("failed to wait for host startup")?;
+
+        deploy_wadm_application(
+            &client,
+            &manifest_path,
+            lattice.as_ref(),
+            &mut out_text,
+            &mut out_json,
+        )
+        .await
+        .with_context(|| {
+            format!(
+                "failed to deploy wadm application @ path [{}]",
+                manifest_path.display()
+            )
+        })?;
+    }
+
     // Write the pid file with the selected version
     tokio::fs::write(install_dir.join(WASMCLOUD_PID_FILE), version).await?;
     if !cmd.detached {
         run_wasmcloud_interactive(&mut wasmcloud_child, output_kind).await?;
 
         let spinner = Spinner::new(&output_kind)?;
+
+        // If a WADM application was specified when we started, shut it down on exit
+        if let Some(ref manifest_path) = cmd.wadm_opts.wadm_application {
+            spinner.update_spinner_message(
+                // wadm and NATS both exit immediately when sent SIGINT
+                "CTRL+c received, undeploying specified WADM manifest...".to_string(),
+            );
+            let manifest = load_app_manifest(AppManifestSource::File(manifest_path.clone()))
+                .await
+                .with_context(|| {
+                    format!(
+                        "failed to load manifest from path [{}]",
```

Comment:
Nit: Might want to note that this happened on cleanup and manual cleanup will be required